packages/core/src/liner.ts

```diff
@@ -58,18 +58,84 @@ export function removeLineNumbers(text: string) {
  * @param options - An object specifying the line range.
  *   - lineStart: The 1-based starting line number of the range.
  *   - lineEnd: The 1-based ending line number of the range.
+ *   - line: Center line number around which to extract content dynamically.
  * @returns The extracted range of text or the original text if no valid range is provided.
  */
-export function extractRange(text: string, options?: { lineStart?: number; lineEnd?: number }) {
-  const { lineStart, lineEnd } = options || {};
-  if (isNaN(lineStart) && isNaN(lineEnd)) return text;
+export function extractRange(text: string, options?: { lineStart?: number; lineEnd?: number; line?: number }) {
+  const { lineStart, lineEnd, line } = options || {};
+  
+  // Handle existing lineStart/lineEnd logic first (takes priority)
+  if (!isNaN(lineStart) || !isNaN(lineEnd)) {
+    const lines = text.split("\n");
+    const startLine = lineStart || 1;
+    const endLine = lineEnd || lines.length;
+    return lines.slice(startLine - 1, endLine).join("\n");
+  }
+  
+  // Handle center line option if lineStart/lineEnd not provided
+  if (!isNaN(line)) {
+    return extractRangeAroundLine(text, line);
+  }
+  
+  // If no valid range is provided, return original text
+  return text;
+}
 
+/**
+ * Extracts a dynamic range around a center line.
+ * The range size is calculated based on file size and context needs.
+ * 
+ * @param text - The input text from which to extract the range.
+ * @param centerLine - The 1-based center line number.
+ * @returns The extracted range of text around the center line.
+ */
+export function extractRangeAroundLine(text: string, centerLine: number): string {
```

Comment:
This function should take a maxTokens budget and compute the length of lines around lines until it hits the budget